extensions/blocks/gathering-tweetstorms/use-gather-tweetstorm.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * External dependencies
+ */
+import { isArray, isEmpty, some } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { addQueryArgs } from '@wordpress/url';
+import { createBlock } from '@wordpress/blocks';
+import { useSelect, dispatch } from '@wordpress/data';
+
+/**
+ * Given a tweet URL, find any tweetstorm replies it has, and replace the current
+ * block with the tweetstorm content.
+ *
+ * @param {object}   o - Function parameters.
+ * @param {Function} o.onReplace - The onReplace() function passed down from the block.
+ *
+ * @returns {object} Object whether the API call is still running, and unleashStorm(),
+ * a function to replace the current block with the tweetstorm content.
+ */
+export default function useGatherTweetstorm( { onReplace } ) {
+	const [ isGatheringStorm, setIsGatheringStorm ] = useState( false );
+
+	const connections = useSelect( select => {
+		return select( 'core/editor' ).getEditedPostAttribute( 'jetpack_publicize_connections' );
+	} );
+
+	/**
+	 * Import the tweet content, and replace the current block with that content.
+	 *
+	 * @param {string}   url - The tweet URL.
+	 * @param {Function} noticeOperations - From WordPress' withNotices() HOC.
+	 */
+	const unleashStorm = ( url, noticeOperations ) => {
+		if ( isEmpty( url ) ) {
+			return;
+		}
+
+		const userResult = url.match( /^https?:\/\/(?:www\.)?twitter\.com\/([^/]+)\/status\/\d+/ );
+		if ( isEmpty( userResult ) ) {
+			return;
+		}
+
+		const twitterUser = userResult[ 1 ];
+
+		noticeOperations.removeAllNotices();
+		setIsGatheringStorm( true );
+
+		apiFetch( {
+			path: addQueryArgs( '/wpcom/v2/tweetstorm/gather', { url } ),
+		} )
+			.then( blocks => {
+				setIsGatheringStorm( false );
+
+				if ( ! isArray( blocks ) ) {
+					noticeOperations.createErrorNotice(
+						__( 'An error occurred. Please try again later.', 'jetpack' )
+					);
+
+					return;
+				}
+
+				if ( isEmpty( blocks ) ) {
+					noticeOperations.createErrorNotice(
+						__( 'We were unable to get any content from this tweet.', 'jetpack' )
+					);
+
+					return;
+				}
+
+				onReplace(
+					blocks.map( block => {
+						switch ( block.type ) {
+							case 'paragraph':
+								return createBlock( 'core/paragraph', { content: block.content } );
+							case 'gallery':
+								return createBlock( 'core/gallery', { images: block.images } );
```

Comment:
This isn't possible without sideloading, since tiled gallery requires images to be in the media library. I agree that it would be a nice upgrade, though.